backend/services/oeg_report_service.py

```diff
@@ -0,0 +1,280 @@
+from flask import current_app
+import requests
+from requests.exceptions import ConnectionError
+import json
+
+from backend import create_app
+from backend.api.utils import TMAPIDecorators
+from backend.models.dtos.project_dto import ProjectDTO
+from backend.models.postgis.project import Project, ProjectStatus
+from backend.models.postgis.licenses import License
+from backend.models.postgis.organisation import Organisation
+from backend.models.postgis.project_info import ProjectInfo
+from backend.models.dtos.oeg_report_dto import (
+    ExternalSourceReportDTO,
+    ProjectReportDTO,
+    OrganisationReportDTO,
+    OegReportDTO,
+)
+
+
+tm = TMAPIDecorators()
+
+
+class OegReportService:
+    @tm.asynchronous()
+    def report_data_to_osm(
+        self, project_dto: ProjectDTO, current_project_dto: ProjectDTO
+    ):
+        app = (
+            create_app()
+        )  # Because message-all run on background thread it needs it's own app context
+
+        with app.app_context():
+            project_id = project_dto.project_id
+            project = Project.get(project_id)
+            organisation_enabled_report = project.organisation.enabled_oeg_report
+
+            # Format project report data
+            project = Project.get(project_id)
```

Comment:
It's executing the same query 2 times here and another one in the project_admin_service.py. Wouldn't it be possible to receive the project model object instead of the DTO?

Reply:
In the `oeg_report_service.py`,  it really was not necessary to have the same query twice, so I removed one of it.

About receiving the `DTO` object instead of the model, it's because I need to check which fields were updated in the Tasking Manager and compare it with the updated model object instantiated in the `oeg_report_service`, this wouldn't be possible if I used the model object as a parameter because it would be already updated with the new project fields values.

It's OK to keep this logic?

Reply:
All right, it's ok!